karapace/schema_registry.py

```diff
@@ -329,24 +342,19 @@ async def write_new_schema_local(
                     )
                     return schema_id
 
-                maybe_schema_id = self.schema_reader.get_schema_id_if_exists(subject=subject, schema=new_schema)
-                if maybe_schema_id is not None:
-                    LOG.debug("Schema id %r found from subject+schema cache", maybe_schema_id)
-                    return maybe_schema_id
-
-                compatibility_mode = self.get_compatibility_mode(subject=subject_data)
+                compatibility_mode = self.get_compatibility_mode(subject=subject)
 
                 # Run a compatibility check between on file schema(s) and the one being submitted now
                 # the check is either towards the latest one or against all previous ones in case of
                 # transitive mode
-                schema_versions = sorted(list(schemas))
+                schema_versions = sorted(list(live_schema_versions))
```

Comment:
```suggestion
                schema_versions = sorted(live_schema_versions)
```

nitpick (non-blocking): There should be no need to instantiate a list here, `sorted` is `(Iterable[T]) -> list[T]` I think which `dict` is compatible with.

Reply:
Done.